app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ApplicationsController < ApplicationController
+    def show
+        @application = Application.find(params[:id])
+        
+        @pets = Pet.search(params[:search])
+        
+        if params[:added_pet].present? 
+            pet_id = params[:added_pet]
+            @application.add_pet_to_application(pet_id)
+        end
+
+        if params[:pet_ownership_description].present?
```

Comment:
For this functionality, consider the "Submit" button routing to a PATCH action in this controller that would update the status of the Application resource. Then it would redirect back to the show page.